src/components/CreatureContainer.js

```diff
@@ -66,7 +66,7 @@ class CreatureContainer extends Component {
     if (this.state.hasError) {
```

Comment:
I like how you are incorporating error handling, and I have not looked at your API calls, but would having a function in your api calls that checks the response before returning the response.json() would do the same as you are doing here? That way you could then in your catch, setState for your `hasError` to true. Let me know if this makes sense for you and I would be happy to talk through it over a call. :)

```
function checkResponse(response) {
   if (!response.ok) {
      throw new Error("Error message")
   }
   return response.json()
}
```